runtime/lua/vim/treesitter/query.lua

```diff
@@ -408,12 +425,21 @@ local directive_handlers = {
 
     -- If this produces an invalid range, we just skip it.
     if range[1] < range[3] or (range[1] == range[3] and range[2] <= range[4]) then
-      if not metadata[capture_id] then
-        metadata[capture_id] = {}
-      end
       metadata[capture_id].range = range
     end
   end,
+
+  -- Transform the content of the node
+  -- Example: (#gsub! @_node ".*%.(.*)" "%1")
+  ['gsub!'] = function(match, _, bufnr, pred, metadata)
+    assert(#pred == 4)
+
+    local id = pred[2]
+    local node = match[id]
+    local text = M.get_node_text(node, bufnr, { metadata = metadata[id] }) or ''
+
+    metadata[id] = text:gsub(pred[3], pred[4])
```

Comment:
Should this behave like `offset!` where we set an explicit field in metadata?

Maybe: 
```lua
metadata[id] = metadata[id] or {}
metadata[id].content = text:gsub(pred[3], pred[4])
```

Reply:
I have now reverted https://github.com/neovim/neovim/pull/21548/commits/b1f803637d47a7265d7bf4bc44c38cda2f05b23b, I was initially under the impression that I should do it the other way around from https://github.com/neovim/neovim/pull/21548#discussion_r1057350981

Reply:
That comment is a bit ambiguous, I didn't read it as store the result *as* the metadata, but *in* the metadata.